persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/stringbased/JdbcStringBasedStore.java

```diff
@@ -306,101 +309,122 @@ static class PossibleExpirationNotification {
-                        purgedAmount += runBatchAndNotify(list, batchDelete, unicastProcessor);
-                        possibleAmount = 0;
+   private void purgeExpiredInBlockingThread(UnicastProcessor<MarshallableEntry<K, V>> unicastProcessor) {
+      try {
+         boolean key2StringMapperIsAvailable = key2StringMapper instanceof TwoWayKey2StringMapper;
+         if (!key2StringMapperIsAvailable) {
+            PERSISTENCE.twoWayKey2StringMapperIsMissing(TwoWayKey2StringMapper.class.getSimpleName());
+         }
+         long purgedAmount = 0L;
+         List<PossibleExpirationNotification> entriesToBeDeleted;
+         int[] deleteResults;  // after call of deleteEntries the same length as entriesToBeDeleted
+         do {
+            entriesToBeDeleted = getEntriesToBeDeleted();
+
+            if (!entriesToBeDeleted.isEmpty()) {
+               deleteResults = deleteEntries(entriesToBeDeleted);
+               if (deleteResults != null) {
+                  if (key2StringMapperIsAvailable) {
+                     if (deleteResults.length == entriesToBeDeleted.size()) {
+                        purgedAmount += doNotify(entriesToBeDeleted, deleteResults, unicastProcessor);
+                     } else {
+                        log.errorf("Expected %d entries to be deleted, but deleteresult contains only %d entries");
                      }
                   }
-
-                  if (list == null || !list.isEmpty()) {
-                     purgedAmount += runBatchAndNotify(list, batchDelete, unicastProcessor);
-                  }
-
-                  if (log.isTraceEnabled()) {
-                     log.tracef("Successfully purged %d rows.", purgedAmount);
-                  }
-                  conn.commit();
-                  unicastProcessor.onComplete();
+               } else {
+                  log.errorf("Expected %d entries to be deleted, but deleteresult is empty");
                }
-            } catch (SQLException e) {
-               log.failedClearingJdbcCacheStore(e);
-               try {
-                  conn.rollback();
-               } catch (SQLException ex) {
-                  log.sqlFailureTxRollback(ex);
+            }
+         } while (!entriesToBeDeleted.isEmpty());
+         if (purgedAmount > 0) {
+            log.infof("Successfully purged %d rows of table %s.", purgedAmount, getTableManager().getDataTableName());
+         }
+         unicastProcessor.onComplete();
+      } catch (SQLException e) {
+         log.failedClearingJdbcCacheStore(e);
+         unicastProcessor.onError(e);
+      }
+   }
+
+   private List<PossibleExpirationNotification> getEntriesToBeDeleted() throws SQLException {
+      int batchSize = this.configuration.maxBatchSize();
+      String sql = this.getTableManager().getSelectOnlyExpiredRowsSql();
+      try (var conn = this.connectionFactory.getConnection()) {
+         conn.setAutoCommit(false);
+         try(PreparedStatement ps = conn.prepareStatement(sql)) {
```

Comment:
Spacing is off in a few other places as well. Can you try using the code style at https://github.com/infinispan/infinispan/tree/main/ide-settings and reformat the files?
```suggestion
         try (PreparedStatement ps = conn.prepareStatement(sql)) {
```

Reply:
Pure copy did no reformat. Should be corrected after commit